### ISSUE TITLE
[Bug] Can't rename files in Shared with me even though I have full access

### ISSUE THREAD

**Issue by tprudentova** [linagora/twake-drive-legacy#639]
1.0.3, staging, Firefox

1. User A shares a file with User B though "Manage access" with permission level "Full access"
2. User B opens the Shared with me section 
3. User B tries to rename the shared file 
4. "Unable to load your files"

Expectation: I can rename files when I have full access

https://github.com/user-attachments/assets/a0764af3-dd18-4d0a-84ea-ae2280322359

**Comment by ericlinagora** [comment]
This is an odd one in terms of rights, for example, say user A is storing HR complaints in `my drive/complaints/b.txt` and then shares  `b.txt` with user B so they can read about it, if user B then renames the file, we have to check there isn't, in user A's complaints folder, already a file named like that, which would allow user B to find out wether there is a complaint against C by trying to rename his file to `c.txt`

**Comment by chibenwa** [comment]
Agreed this "feature" can lead to **information disclosure**.

I think here the issue arise from an unclear error handling. How about being clearer?

 - `We are unable to rename this file as you miss acces rights` on the parent folder
 - Also  if I have edition right on the parent I expect rename to succeed...

**Comment by tprudentova** [comment]
Since this problem applies to basically any shared file/folder, maybe we should simply remove the "rename" option from the action menu?

**Comment by ericlinagora** [comment]
I think the option of doing it when we have at least read rights on the parent folder works, with indeed a clearer message to explain what's up, it's as you wish

**Comment by tprudentova** [comment]
So, basically, we can do the following, right?

Folder tree: My Drive/Folder A/file.txt

Case 1: only file.txt is shared with me

Suggested behaviour: 

-	I try to rename file.txt 
-	I see a warning “You don’t have enough rights to rename this file” 

Case 2: Folder A is also shared with me

Suggested behaviour for file.txt: 

-	I try to rename file.txt 
-	I can rename file.txt, yay! 

Suggested behaviour for Folder A: 

-	I try to rename Folder A 
-	I see a warning “You don’t have enough rights to rename this folder” (because its parent folder is My Drive itself)

**Comment by ericlinagora** [comment]
I think that works, but folder A wouldn't be a special case, however we really need to make an explicit message for this instance because there is no way any user could guess they need read access to the parent to do it, it's very counter-intuitive. Something like "You cannot rename a file/folder that is shared with you if you do not have read access to the parent folder of the file/folder (of the file's creator)"

**Comment by shepilov** [comment]
guys, if the file is shared with "manage" permissions, it means user B has the same permissions as the owner of the file.
If you want to share complaints, you can share it with "read" or "write" permissions. And there is no need to check permission on the parent folder

**Comment by ericlinagora** [comment]
what if you rename the file to a name that already exists in the parent folder then ?

**Comment by shepilov** [comment]
@ericlinagora Yes, let's check "read" permissions for the parent folder for "rename" action
